### PR TITLE
Amazon Personalize support on Langchain_experimental

### PR DESCRIPTION
**Pull request by pranava-amzn**
## Amazon Personalize support on Langchain

This PR is a successor to this PR - https://github.com/langchain-ai/langchain/pull/13216 

This PR introduces an integration with [Amazon Personalize](https://aws.amazon.com/personalize/) to help you to retrieve recommendations and use them in your natural language applications. This integration provides two new components:

1. An `AmazonPersonalize` client, that provides a wrapper around the Amazon Personalize API.
2. An `AmazonPersonalizeChain`, that provides a chain to pull in recommendations using the client, and then generating the response in natural language. 

We have added this to langchain_experimental since there was feedback from the previous PR about having this support in experimental rather than the core or community extensions. 

Here is some sample code to explain the usage.

```python

from langchain_experimental.recommenders import AmazonPersonalize
from langchain_experimental.recommenders import AmazonPersonalizeChain
from langchain.llms.bedrock import Bedrock

recommender_arn = "<insert_arn>"

client=AmazonPersonalize(
    credentials_profile_name="default",
    region_name="us-west-2",
    recommender_arn=recommender_arn
)
bedrock_llm = Bedrock(
    model_id="anthropic.claude-v2", 
    region_name="us-west-2"
)

chain = AmazonPersonalizeChain.from_llm(
    llm=bedrock_llm, 
    client=client
)
response = chain({'user_id': '1'})
```

Reviewer: @3coins

### CODE REVIEW

**Comment by pranava-amzn** [comment]
@3coins - please review and let me know if there is any feedback

File: cookbook/amazon_personalize_how_to.ipynb [hunk 1]
```diff
@@ -0,0 +1,284 @@
...
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "collapsed": false,
+    "jupyter": {
+     "outputs_hidden": false
+    }
+   },
+   "source": [
+    "### 2.6 [Use-Case 6] Setup Amazon Personalize Client and invoke Personalize Chain with returned metadata for summarizing results"
```
Comment (3coins) [inline cookbook/amazon_personalize_how_to.ipynb:232]: Should this be "Invoke Personalize Chain..."?

**Review by 3coins — COMMENTED** [review]
@pranava-amzn 
Left some comments for the included sample notebook. Looks good otherwise.

File: cookbook/amazon_personalize_how_to.ipynb [hunk 2]
```diff
@@ -0,0 +1,284 @@
...
+    "recommender_arn = \"<insert_arn>\"\n",
+    "\n",
+    "client = AmazonPersonalize(\n",
+    "    credentials_profile_name=\"default\",\n",
+    "    region_name=\"us-west-2\",\n",
+    "    recommender_arn=recommender_arn,\n",
+    ")\n",
+    "client.get_recommendations(user_id=\"1\")"
+   ]
+  },
```
[the remark this answers is not in the document]
Comment (pranava-amzn) [reply cookbook/amazon_personalize_how_to.ipynb]: Done

File: cookbook/amazon_personalize_how_to.ipynb (hunk 1 above)
Comment (pranava-amzn) [reply cookbook/amazon_personalize_how_to.ipynb]: Done

File: cookbook/amazon_personalize_how_to.ipynb [hunk 3]
```diff
@@ -0,0 +1,284 @@
...
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "collapsed": false,
+    "jupyter": {
+     "outputs_hidden": false
+    }
+   },
+   "source": [
+    "### 2.2 [Use-case-2] Setup Amazon Personalize Client and invoke Personalize Chain for summarizing results"
```
[the remark this answers is not in the document]
Comment (pranava-amzn) [reply cookbook/amazon_personalize_how_to.ipynb]: Done

**Comment by pranava-amzn** [comment]
> @pranava-amzn Left some comments for the included sample notebook. Looks good otherwise.

Done. Thanks for pointing them out. I have updated the notebook.

**Review by 3coins — APPROVED** [review]
@pranava-amzn 
Looks good. Thanks for the updates.

**Comment by 3coins** [comment]
@baskaryan 
This is a refactor for PR [#13216](https://github.com/langchain-ai/langchain/pull/13216) to move the code to experimental. Let us know if anything else is needed here. Thanks.

**Pull request merged**